ecosystem/typescript/sdk_v2/src/internal/account.ts

```diff
@@ -121,3 +122,19 @@ export async function getResource(args: {
   });
   return data;
 }
+
+export async function estimateAccountMaxGasAmount(args: { aptosConfig: AptosConfig; accountAddress: HexInput }) {
+  const { aptosConfig, accountAddress } = args;
+  // Only APT coin is accepted as gas
+  const typeTag = `0x1::coin::CoinStore<${APTOS_COIN}>`;
+  const [{ gas_estimate: gasUnitPrice }, resources] = await Promise.all([
+    getGasPriceEstimation({ aptosConfig }),
+    getResources({ aptosConfig, accountAddress }),
+  ]);
+  const accountResource = resources.find((r) => r.type === typeTag);
+  if (!accountResource) {
+    throw new Error(`account ${accountAddress} doesnt have a ${typeTag} resource`);
+  }
+  const balance = BigInt((accountResource.data as any).coin.value);
+  return balance / BigInt(gasUnitPrice);
```

Comment:
I think I'm confused what this number is supposed to represent. MaxGasAmount is specific to a transaction, and comes out of simulation.  I'm not sure we need this function

Reason for this is, if you want the `Max Max gas amount`, you'll need to retrieve the gas schedule, and ceiling it to the maximum gas value, which last I remember is 2 million gas units.

See: https://github.com/aptos-labs/aptos-core/blob/main/api/src/transactions.rs#L452-L510